app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -1,18 +1,43 @@
 package com.andrew.tasky.agenda.data.event
 
+import androidx.core.net.toFile
 import com.andrew.tasky.agenda.data.database.AgendaDatabase
 import com.andrew.tasky.agenda.domain.EventRepository
 import com.andrew.tasky.agenda.domain.models.AgendaItem
 import com.andrew.tasky.agenda.domain.models.Attendee
+import com.andrew.tasky.agenda.domain.models.EventPhoto
 import com.andrew.tasky.auth.data.AuthResult
+import com.andrew.tasky.auth.util.getAuthResult
 import javax.inject.Inject
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import okhttp3.MultipartBody
+import okhttp3.RequestBody.Companion.toRequestBody
 
 class EventRepositoryImpl @Inject constructor(
-    private val db: AgendaDatabase
+    private val db: AgendaDatabase,
+    private val api: EventApi
 ) : EventRepository {
 
-    override suspend fun createEvent(): AuthResult<Unit> {
-        TODO("Not yet implemented")
+    override suspend fun createEvent(event: AgendaItem.Event) {
+        getAuthResult {
+            api.createEvent(
+                eventData = MultipartBody.Part
+                    .create(Json.encodeToString(event.toCreateEventRequest()).toRequestBody()),
+                photoData = event.photos?.map { eventPhoto ->
+                    when (eventPhoto) {
+                        is EventPhoto.Local -> {
+                            MultipartBody.Part
+                                .create(eventPhoto.uri.toFile().readBytes().toRequestBody())
+                        }
+                        is EventPhoto.Remote -> {
+                            MultipartBody.Part
+                                .create(eventPhoto.photoUrl.encodeToByteArray().toRequestBody())
+                        }
+                    }
+                } ?: emptyList()
```

Comment:
1. You're correctly attaching the photos and the JSON, however you don't specify any names for it. The API needs to know what each attachment represents, for which you can attach the key like `create_event_request`. For the photos it's `photo0`, `photo1`, ..., `photo9`
2. There's no point in uploading remote photos again as they are already on the server (or here for creating events there's no scenario where there would even be a remote photo). You can therefore safely just filter for local photos using `event.photos.filterIsInstance<EventPhoto.Local>`